pkg/apis/planetscale/v2/vitessbackupschedule_types.go

```diff
@@ -0,0 +1,206 @@
+
+	// SuccessfulJobsHistoryLimit defines how many successful jobs will be kept around.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty"`
+
+	// FailedJobsHistoryLimit defines how many failed jobs will be kept around.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty"`
+
+	// Suspend enables the suspension of the VitessBackupSchedule, pausing any further scheduled
+	// runs until Suspend is set to false again. This is useful if you want to pause backup without
+	// having to remove the entire VitessBackupSchedule object from the cluster.
+	// +optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// StartingDeadlineSeconds allows for the VitessBackupSchedule controller to start jobs that late
+	// by the given amount of seconds.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
+
+	// ConcurrencyPolicy specifies ho to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Allow" (default): allows CronJobs to run concurrently;
+	// - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it with a new one.
+	// +optional
+	// +kubebuilder:example="Forbid"
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
+
+	// AllowedMissedRun defines how many missed run of the schedule will be allowed before giving up on finding the last job.
+	// If the operator's clocked is skewed and we end-up missing a certain number of jobs, finding the last
+	// job might be very consuming, depending on the frequency of the schedule and the duration during which
+	// the operator's clock was misbehaving, also depending on how laggy the clock is, we can end-up with thousands
+	// or missed runs. For this reason, AllowedMissedRun, which is set to 100 by default, will allow us to give up finding
+	// the last job and simply wait for the next job on the schedule.
+	// Unless you are experiencing issue with missed runs due to a misconfiguration of the clock, we recommend leaving
+	// this field to its default value.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	AllowedMissedRun *int `json:"allowedMissedRun,omitempty"`
+
+	// JobTimeoutMinute defines after how many minutes a job that has not yet finished should be stopped and removed.
+	// Default value is 10 minutes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=10
+	JobTimeoutMinute int32 `json:"jobTimeoutMinute,omitempty"`
+}
+
+// VitessBackupScheduleStrategy defines how we are going to take a backup.
+// The VitessBackupSchedule controller uses this data to build the vtctldclient
+// command line that will be executed in the Job's pod.
+type VitessBackupScheduleStrategy struct {
```

Comment:
it seems like it would be kind of tedious to define a strategy for every shard in the cluster. i imagine a pretty common use case would be "take a backup of every shard in the cluster". might be nice if there was a third strategy "backup cluster".